gdx/src/com/badlogic/gdx/utils/ArrayMap.java

```diff
@@ -398,6 +398,38 @@ public void truncate (int newSize) {
 		size = newSize;
 	}
 
+	@Override
+	public int hashCode () {
+		int h = 0;
+		for (Entry<K, V> entry : entries()) {
+			h += (entry.key != null ? entry.key.hashCode() : 0) * 31;
+			h += (entry.value != null ? entry.value.hashCode() : 0);

```

Comment:
Should be multiplied by a prime. Both prime multiplications can be avoided when null.